File: app/src/main/java/com/shindra/LineAdapter.java
package com.shindra;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.shindra.ctslibrary.bo.Line;

import java.util.ArrayList;

public class LineAdapter extends RecyclerView.Adapter<LineViewHolder>{

    //int pictures[];
    private ArrayList<Line> lines;
    private LineClick callback;

    public LineAdapter(ArrayList<Line> lines, LineClick callback){
        //this.pictures = pictures;
        this.lines = lines;
        this.callback = callback;
    }

    @NonNull
    @Override
    public LineViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.lineview, parent, false);
        return new LineViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull LineViewHolder holder, int position) {
        //holder.onBind(pictures[position]);
        Line lines = this.lines.get(position);
        holder.onBind(lines, callback);
    }

    @Override
    public int getItemCount() {
        //return pictures.length;
        return lines.size();
    }
}

File: app/src/main/java/com/shindra/LineAdapter.kt
package com.shindra

import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import com.shindra.ctslibrary.bo.Line
import java.util.*

class LineAdapter(private val lines: ArrayList<Line>, val callback: LineViewHolder.OnLineClickListener) : RecyclerView.Adapter<LineViewHolder>() {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): LineViewHolder {
        val lineView = LayoutInflater.from(parent.context).inflate(R.layout.line_card_view, parent, false)
        return LineViewHolder(lineView)
    }

    override fun onBindViewHolder(holder: LineViewHolder, position: Int) {
        val line = lines[position]
        holder.tramIcon.setImageResource(getTramIcon(line))
        holder.onBind(line,callback)
    }

    override fun getItemCount(): Int {
        return lines.size
    }

    private fun getTramIcon(line: Line) : Int {
        return when (line.name){
            "Ligne A" -> R.drawable.tram_a
            "Ligne B" -> R.drawable.tram_b
            "Ligne C" -> R.drawable.tram_c
            "Ligne D" -> R.drawable.tram_d
            "Ligne E" -> R.drawable.tram_e
            "Ligne F" -> R.drawable.tram_f
            else -> R.drawable.tram
        }
    }
}